### PR TITLE
BOT-657 Add Ability to Track audio player state, send device capabilities and send audioPlayer intents

### PR DESCRIPTION
**Pull request by StoutyAlex**
References: https://github.com/codeforequity-at/botium-core/issues/312

This PR contains changes mainly to the Invocation section of the connector. 

The reason for this  is because I wanted to add the ability to send AudioPlayer intents which can't really be specified in simulation, such as `AudioPlayer.PlaybackNearlyFinished` and other AudioPlayer intents. 

### New Assertions
When receiving a media response with an AudioPlayer directive we can now check that the media uri is correct using the media tag
``` MEDIA www.example.com/test.mp3```

#### ALEXA_SMAPI_AUDIO_CAPABILITY and ALEXA_SMAPI_DISPLAY_CAPABILITY
These when set to `true` will add audio / display capabilities to the requestTemplate. This can be done on both simulation and invocation

#### ALEXA_SMAPI_REFRESH_USER_ID
When set to true a new userId will be generated every conv.txt file for cases when you want to test databases on the skill that involve a userId such as playing audio files a max of 3 times for regular users. This only works with invocation since it updates the invocationRequest object

#### ALEXA_SMAPI_KEEP_AUDIO_PLAYER_STATE
This is the big change in this PR and will allow changes on the audioPlayer when botium receives an AudioPlayer response / directive. it will adapt the invocation template with the audioPlayers event. 
So it can set the `token`, `offsetInMilliseconds` and the `playerActivity` The only audio player events this supports at the moment is 'PLAY' and 'STOP'
This only works with invocation since it updates the invocationRequest object

### CODE REVIEW

**Comment by codeforequity-at** [comment]
Thanks, will review it in the next days ...

File: index.js [hunk 1]
```diff
@@ -30,6 +40,13 @@ class BotiumConnectorAlexaSmapi {
     this.locale = this.caps['ALEXA_SMAPI_LOCALE'] || 'en-US'
     this.endpointRegion = this.caps['ALEXA_SMAPI_ENDPOINTREGION'] || 'default'
 
+    this.audioCapability = this.caps['ALEXA_SMAPI_AUDIO_CAPABILITY'] === true
```
Comment (codeforequity-at) [inline index.js]: Maybe better to use:

    this.audioCapability = !!this.caps['ALEXA_SMAPI_AUDIO_CAPABILITY']

File: index.js [hunk 2]
```diff
@@ -105,8 +122,25 @@ class BotiumConnectorAlexaSmapi {
                   debug(`got simulation request: ${JSON.stringify(simulationRequest)}`)
                   const simulationResult = askTools.convertDataToJsonObject(response.result.skillExecutionInfo.invocationResponse.body.response)
                   debug(`got simulation result: ${JSON.stringify(simulationResult)}`)
-                  const messageText = simulationResult.outputSpeech.text || simulationResult.outputSpeech.ssml
-                  const botMsg = { sender: 'bot', sourceData: simulationResult, messageText }
+
+                  let messageText = 'no text response from skill'
```
Comment (codeforequity-at) [inline index.js]: a default text if no response ? looks a little bit fishy. If there is a good reason why to add a default text here, then maybe as additional configuration capability.

Comment (StoutyAlex) [reply index.js]: There was no reason, this was the first change I made and did it this way to understand how it works. Have removed it now

File: index.js (hunk 1 above)
Comment (StoutyAlex) [reply index.js]: Good idea didn't know about the double `!!` Thank you!

**Comment by StoutyAlex** [comment]
How long till this will be merged?

**Comment by codeforequity-at** [comment]
next build on monday ... timing was bad, pushed right after the last build ...

**Comment by codeforequity-at** [comment]
It is available in the latest Botium builds now (including Botium CLI and Botium Box)

**Comment by StoutyAlex** [comment]
Thank you! I will check out the discord channel at a later date :)

**Pull request merged**